packages/contracts-bedrock/justfile

```diff
@@ -293,16 +292,40 @@ check:
 ########################################################
 
 # Cleans, builds, lints, and runs all checks.
-pre-pr: clean pre-pr-no-build
-
-# Builds, lints, and runs all checks. Sometimes a bad cache causes issues, in which case the above
-# `pre-pr` is preferred. But in most cases this will be sufficient and much faster then a full build.
-# Steps:
-# - Build (dev mode) to confirm that contracts compile
-# - Lint contracts
-# - Generate snapshots
-# - Run all checks to confirm everything is good
-pre-pr-no-build: build-dev lint snapshots-no-build check
+# Alias for pre-pr.
+pre-commit *ARGS:
+  just pre-pr {{ARGS}}
+
+# Cleans, builds, lints, and runs all checks.
+pre-pr *ARGS:
+  #!/bin/bash
+  # Optionally clean the previous build.
+  # --clean is typically not needed but can force a clean build if you suspect cache issues.
+  if [[ "{{ARGS}}" == *"--clean"* ]]; then
+      just clean
+  fi
+
+  # Create temp directory for build cache if it doesn't exist
+  TEMP_BUILD_DIR=$(mktemp -d)
+  trap '[ -d "$TEMP_BUILD_DIR" ] && rm -rf "$TEMP_BUILD_DIR"' EXIT INT
+
+  just build-dev
+
+  # Cache build artifacts
+  cp -r artifacts "$TEMP_BUILD_DIR/"
+  cp -r forge-artifacts "$TEMP_BUILD_DIR/"
+  cp -r cache "$TEMP_BUILD_DIR/"
```

Comment:
Why do we build, then cache that build? Can you add some comments to explain?

Reply:
Added a comment for this